community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/AllRelationshipIterator.java

```diff
@@ -38,4 +44,25 @@
         visitor.visit( relationshipId, record.getType(), record.getFirstNode(), record.getSecondNode() );
         return false;
     }
+
+    @Override
+    protected boolean doFetchNext( long highId )
+    {
+        while ( currentId <= highId )
+        {
+            try
+            {
+                store.getRecord( currentId, record, RecordLoad.CHECK );
+                if ( record.inUse() )
+                {
+                    return next( record.getId() );
+                }
```

Comment:
That would be nice, but you need the record in `relationshipVisit` which is called on each element in the iterator.